chains/evm/contracts/pools/USDC/USDCTokenPoolProxy.sol

```diff
@@ -0,0 +1,378 @@
+    if (i_router.getOnRamp(lockOrBurnIn.remoteChainSelector) != msg.sender) {
+      revert Unauthorized();
+    }
+
+    LockOrBurnMechanism mechanism = s_lockOrBurnMechanism[lockOrBurnIn.remoteChainSelector];
+
+    // If a mechanism has not been configured for the remote chain selector, revert.
+    if (mechanism == LockOrBurnMechanism.INVALID_MECHANISM) {
+      revert InvalidLockOrBurnMechanism(mechanism);
+    }
+
+    PoolAddresses memory pools = s_pools;
+
+    address destinationPool;
+
+    // The order of the branches is based on the expected frequency of each mechanism being used, in order to save
+    // gas on every call.
+    if (mechanism == LockOrBurnMechanism.CCTP_V2) {
+      destinationPool = pools.cctpV2Pool;
+    } else if (mechanism == LockOrBurnMechanism.CCTP_V1) {
+      destinationPool = pools.cctpV1Pool;
+    } else if (mechanism == LockOrBurnMechanism.LOCK_RELEASE) {
+      destinationPool = s_lockReleasePools[lockOrBurnIn.remoteChainSelector];
+    }
+
+    // If the destination pool is the zero address, then no mechanism has been configured for the outgoing tokens
+    // and thus the remote chain is not supported and should revert.
+    if (destinationPool == address(0)) {
+      revert InvalidDestinationPool();
+    }
+
+    // Transfer the tokens to the proper child pool, as this contract is only a proxy and will not perform
+    // the lock/burn itself.
+    i_token.safeTransfer(destinationPool, lockOrBurnIn.amount);
+
+    return USDCTokenPool(destinationPool).lockOrBurn(lockOrBurnIn);
+  }
+
+  /// @inheritdoc IPoolV1
+  function isSupportedChain(
+    uint64 remoteChainSelector
+  ) external view returns (bool) {
+    // If the outgoing mechanism is not set for a chain, then the chain is not supported because there cannot be a lock
+    // or burn operation.
+    return s_lockOrBurnMechanism[remoteChainSelector] != LockOrBurnMechanism.INVALID_MECHANISM;
+  }
+
+  /// @inheritdoc IPoolV1
+  function isSupportedToken(
+    address token
+  ) external view returns (bool) {
+    return address(i_token) == token;
+  }
+
+  /// @notice Signals which version of the pool interface is supported
+  function supportsInterface(
+    bytes4 interfaceId
+  ) public pure override returns (bool) {
+    return interfaceId == type(IPoolV1).interfaceId || interfaceId == type(IERC165).interfaceId;
+  }
+
+  /// @inheritdoc IPoolV1
+  function releaseOrMint(
+    Pool.ReleaseOrMintInV1 calldata releaseOrMintIn
+  ) public virtual returns (Pool.ReleaseOrMintOutV1 memory) {
+    // Since this proxy does not inherit from the TokenPool contract, it must manually validate the caller as an offRamp.
+    if (!i_router.isOffRamp(releaseOrMintIn.remoteChainSelector, msg.sender)) {
+      revert Unauthorized();
+    }
+
+    // In previous versions of the USDC Token Pool, the sourcePoolData only contained two fields, a uint64 and uint32.
```

Comment:
This can now be tag based right? At least for CCTPv2 and LockRelease, as those should always send with valid tags. CCTPv1 needs legacy code but that should be fine, at least we'd future proof the others